src/datumaro/components/merge/extractor_merger.py

```diff
@@ -61,23 +64,27 @@ def categories(self) -> CategoriesInfo:
         return self._categories
 
     def __iter__(self) -> Iterator[DatasetItem]:
-        for sources in self._subsets.values():
-            for source in sources:
-                yield from source
+        for subset in self._subsets.values():
+            yield from subset
+
+    def get_subset(self, name: str):
+        if name not in self._subsets:
+            raise KeyError(
+                "Unknown subset '%s', available subsets: %s" % (name, set(self._subsets))
+            )
+        return self._subsets[name]
 
     def __len__(self) -> int:
-        return sum(len(source) for sources in self._subsets.values() for source in sources)
+        return sum(len(subset) for subset in self._subsets.values())
 
     def get(self, id: str, subset: Optional[str] = None) -> Optional[DatasetItem]:
-        if subset is not None and (sources := self._subsets.get(subset, [])):
-            for source in sources:
-                if item := source.get(id, subset):
-                    return item
-
-        for sources in self._subsets.values():
-            for source in sources:
-                if item := source.get(id=id, subset=source.subset):
-                    return item
+        if subset is not None and (source := self._subsets.get(subset)):
+            if item := source.get(id, subset):
+                return item
+
+        for source in self._subsets.values():
+            if item := source.get(id=id, subset=source.subset):
+                return item
```

Comment:
This logic with subset search for `subset == None` handling could be a cool change, but at the moment it's different from other `get` implementations in `Dataset` and `_DatasetBase`. Could you please align them?